### PR TITLE
feat(faceted/badge): BadgePeriod

### PR DESCRIPTION
**Pull request by yyanwang**
**What is the problem this PR is trying to solve?**
Add BadgePeriod. 
[Demo](http://5102.talend.surge.sh/storybook-one/?path=/story/faceted-search-main--default)
[Jira: Period filter on operation page](https://jira.talendforge.org/browse/TDOPS-4573)
[Figma](https://www.figma.com/file/qOdHH37bsaGEtZDo72Osee/TDOPS-4573---Period-filter-for-Operations-page?type=design&node-id=612-97675&mode=design&t=ENYglCvyuKBcZSEW-0)
image

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [dirty]);
+	return (
+		<Form id={`${badgePeriodFormId}-form`} onSubmit={onSubmit}>
+			{!isCustom && (
+				<StackVertical gap="0" align="normal">
+					{getPeriodOptions(t).map(rowItem => {
+						return (
+							<ButtonTertiary
```
Comment (tlnd-mhuchet) [inline packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js:79]: I think the Badge designed in Coral here https://www.figma.com/file/3YWRmMgPIjAABxJl9X9B3W/Badge?type=design&node-id=1112-42165&mode=dev
would be a better fit here (but the implem is not done).

**Review by tlnd-mhuchet — APPROVED** [review]

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js (hunk 1 above)
Comment (jmainguytalend) [reply packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js:79]: 👍

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		if (isDateRange(value)) {
+			if (isValid(value.startDateTime) && isValid(value.endDateTime)) {
+				return getRangeLabel(value.startDateTime, value.endDateTime, t);
+			}
+			return labelAll;
+		}
+		return value.label;
+	}
+	return labelAll;
+};
```
Comment (jmainguytalend) [inline packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js:41]: I would like to suggest following thing as it is more readable and maintainable (only two "return", no if cascade)

```suggestion
const getSelectBadgeLabel = (value, t) => {
	const labelAll = t('FACETED_SEARCH_VALUE_ALL', { defaultValue: 'All' });

	if (isEmpty(value) || !isValid(value.startDateTime) || !isValid(value.endDateTime) return labelAll;
	
	return isDateRange(value)
		? getRangeLabel(value.startDateTime, value.endDateTime, t)
		: value.label;
};
```

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	removable,
+	operator,
+	operators,
+	size,
+	value,
+	values,
+	t,
+	...rest
+}) => {
+	const currentOperators = useMemo(() => operators, [operators]);
```
Comment (jmainguytalend) [inline packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js:61]: useMemo seems useless here, don't you think ?

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+								key={rowItem.id}
+								onClick={event => {
```
Comment (jmainguytalend) [inline packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js:81]: Maybe use hook useCallback could be good optimization for this event listener building

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+const BadgePeriodForm = ({ id, onChange, onSubmit, t, value, ...rest }) => {
+	const [isCustom, setCustom] = useState(value.id === 'CUSTOM');
+	const badgePeriodFormId = `${id}-period-form`;
+	const goBack = () => setCustom(false);
+	const [dirty, setDirty] = useState(false);
+	const initialStartDateTime = subDays(new Date(), 1);
+	const initialEndDateTime = new Date();
+
+	const resetRange = () => {
```
Comment (jmainguytalend) [inline packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js:58]: Maybe use hook useCallback could be good optimization for this event listener

**Review by jmainguytalend — APPROVED** [review]

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js (hunk 5 above)
Comment (yyanwang) [reply packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js:58]: sure, apply `useCallback` here.

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js (hunk 2 above)
Comment (yyanwang) [reply packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js:41]: Simply the function to have less `if` conditions.

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js (hunk 3 above)
Comment (yyanwang) [reply packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriod.component.js:61]: removed the usage of useMemo here.

File: packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js (hunk 1 above)
Comment (yyanwang) [reply packages/faceted-search/src/components/Badges/BadgePeriod/BadgePeriodForm.component.js:79]: thanks for info!
Later we can do a PR to replace the Badge of react-components with the one from design system later when it's ready.

**Pull request merged**